cmd/runtimetest/main.go

```diff
@@ -372,36 +530,46 @@ func validateRootfsPropagation(spec *rspec.Spec, t *tap.T) error {
 			return err
 		}
 		defer unix.Unmount(mountDir, unix.MNT_DETACH)
-		if _, err := os.Stat(filepath.Join(targetDir, filepath.Join(mountDir, filepath.Base(tmpfile.Name())))); os.IsNotExist(err) {
-			if spec.Linux.RootfsPropagation == "shared" {
-				return fmt.Errorf("rootfs should be %s, but not", spec.Linux.RootfsPropagation)
-			}
-			return nil
+		targetFile := filepath.Join(targetDir, filepath.Join(mountDir, filepath.Base(tmpfile.Name())))
+		var exposed bool
+		_, err = os.Stat(targetFile)
+		if os.IsNotExist(err) {
+			exposed = false
+		} else if err != nil {
+			return err
+		} else {
+			exposed = true
 		}
 		if spec.Linux.RootfsPropagation == "shared" {
-			return nil
+			c.harness.Ok(exposed, fmt.Sprintf("shared root propogation exposes %q", targetFile))
+		} else {
```

Comment:
I find the [spec](https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#rootfs-mount-propagation) quite weak about this RootfsPropagation:
- a mount could be both shared and slave
- `rootfsPropagation=slave` is only meaningful if we say what is the master of that enslaved rootfs. Should it necessarily be wrt to a mount in the host namespace?

But for the purpose of this PR, your change is fine.

Reply:
> Should it necessarily be wrt to a mount in the host namespace?

This is another case where we can work around `runtimetest` limitations by limiting the configs we create in `validation/` to things `runtimetest` can check for ;).

> But for the purpose of this PR, your change is fine.

Yeah, I'm fine making not growing this PR to fix everything ;).  And if folks want me to spin off anything I *am* doing (like padjusting the readonly tests][1]), let me know and I'll try to spin those out into their own PRs.

[1]: https://github.com/opencontainers/runtime-tools/pull/308#discussion_r179808463